### DISCUSSION TITLE
Harbor configuration through scripts

### DISCUSSION THREAD

**Discussion by mephmanx** [goharbor/harbor#15132] [General]
All,

I am building an automated infrastructure setup for my personal projects and would like to know if there is a way to setup a repository connection through the harbor api or through postman so I can automate a proxy-cache project through a bash script.  Anyone know?

**Comment by dioguerra** [comment]
You can try and do some reverse WebBrowser engineering and check witch calls and with which data they are created. 
Do this manually and get the full setup curl commands and then convert it to your script. That should do the trick for what you want

**Reply by mephmanx** [reply]
Trying that but I seem to get "error loading resource" when viewing the requests through a developer console.  Happens both in chrome and safari.  I was able to get the scripts together to create a new registry connection, delete the default one, and create a new one but I have 2 problems; 1) the registry connection script fails with a 'registry is unhealthy' error when run through curl but runs fine through postman, and 2) I havent found the tags for the proxy cache setting the the project create script.

**Reply by mephmanx** [reply]
I got this resolved.  Tooka. bit of work but I figured out the API's to do what I needed to do.  Thank you for looking.
